tests/test_utils_schedule_operations.py

```diff
@@ -1,31 +1,217 @@
+                        'invalid_stages': ['not_has_valid_routes'],
+                        'has_valid_routes': False, 'invalid_routes': ['service_0', 'service_1']}},
+        'route_level': {'service': {'service_0': {'is_valid_route': False,
+                                                  'invalid_stages': ['not_has_correctly_ordered_route',
+                                                                     'has_self_loops']},
+                                    'service_1': {'is_valid_route': False,
+                                                  'invalid_stages': ['not_has_correctly_ordered_route',
+                                                                     'has_self_loops']}}},
+        'vehicle_level': {
+            'vehicle_definitions_valid': False,
+            'missing_vehicle_types': {'bus'},
+            'vehicles_affected': {'veh_2_bus': {'type': 'bus'}, 'veh_1_bus': {'type': 'bus'}},
+            'unused_vehicles': set(),
+            'multiple_use_vehicles': {}}}
+
+    report = schedule_validation.generate_validation_report(schedule_with_missing_vehicle_information)
+    assert_semantically_equal(report, correct_report)
+
+
+
+def test_schedule_with_no_unused_vehicles(correct_schedule):
+    correct_output = set()
+    actual_output = correct_schedule.unused_vehicles()
+
+    assert_semantically_equal(correct_output, actual_output)
+
+
+@pytest.fixture()
+def schedule_with_unused_vehicles():
+    s = Schedule(epsg='epsg:27700', services=[
+        Service(id='service',
+                routes=[
+                    Route(route_short_name='route', mode='bus',
+                          stops=[Stop(id='0', x=528504.1342843144, y=182155.7435136598, epsg='epsg:27700'),
+                                 Stop(id='0', x=528504.1342843144, y=182155.7435136598, epsg='epsg:27700')],
+                          trips={'trip_id': ['VJ00938baa194cee94700312812d208fe79f3297ee_04:40:00'],
+                                 'trip_departure_time': ['04:40:00'],
+                                 'vehicle_id': ['veh_1_bus']},
+                          arrival_offsets=['00:00:00', '00:02:00'],
+                          departure_offsets=['00:00:00', '00:02:00']),
+                    Route(route_short_name='route1', mode='bus',
+                          stops=[Stop(id='0', x=528504.1342843144, y=182155.7435136598, epsg='epsg:27700'),
+                                 Stop(id='0', x=528504.1342843144, y=182155.7435136598, epsg='epsg:27700')],
+                          trips={'trip_id': ['Blep_04:40:00'],
+                                 'trip_departure_time': ['05:40:00'],
+                                 'vehicle_id': ['veh_2_bus']},
+                          arrival_offsets=['00:00:00', '00:03:00'],
+                          departure_offsets=['00:00:00', '00:05:00'])
+                ])
+    ])
+    # s.vehicles.pop('veh_2_bus')
+    return s
+
+
+@pytest.mark.skip(reason="TBC")
```

Comment:
What's going on here? 😄 

Skipped tests have a way of hanging around forever if you're not careful. What will lead to us "unskipping" this test?

Reply:
Was waiting for Kasia to take a look - fixed now!